### PR TITLE
Respect CUDA_VISIBLE_DEVICES on worker

### PR DESCRIPTION
**Pull request by kashizui**
Ran this job on a worker with `CUDA_VISIBLE_DEVICES=0`:
https://codalab.stanford.edu/bundles/0xe18d8529909d49c08c02e881d04f8cc3/
And this one on a worker with `CUDA_VISIBLE_DEVICES=2,1`:
https://codalab.stanford.edu/bundles/0x8c56b42ee3344433accae92f8111cc61/

Fixes codalab/codalab-worksheets#220

### CODE REVIEW

**Comment by kashizui** [comment]
Should add documentation to the wiki for this as well.

**Comment by kashizui** [comment]
@percyliang please review :)

File: worker/docker_client.py [hunk 1]
```diff
@@ -123,8 +134,18 @@ def _init_libcuda(self):
...
+                                  m.group(1) in self._cuda_visible_devices):
                 self._nvidia_device_files.append(os.path.join('/dev', filename))
+                if self._cuda_visible_devices is not None:
+                    self._cuda_visible_devices.remove(m.group(1))
+
+        # Check that all requested devices are used
+        if self._cuda_visible_devices is not None and \
+                len(self._cuda_visible_devices) > 0:
+            raise DockerException('Invalid GPU index: ' +
+                                  self._cuda_visible_devices.pop())
```
Comment (percyliang) [inline worker/docker_client.py]: Why not print all of them?

File: worker/docker_client.py [hunk 2]
```diff
@@ -123,8 +134,18 @@ def _init_libcuda(self):
         # Find all the NVIDIA device files.
         self._nvidia_device_files = []
         for filename in os.listdir('/dev'):
-            if filename.startswith('nvidia'):
+            m = re.match(r'nvidia(.*)', filename)
+            if m is not None and (self._cuda_visible_devices is None or
```
Comment (percyliang) [inline worker/docker_client.py]: For clarity, define `device = m.group(1)`, and maybe match `(\d+)` instead of `(.*)`

**Review by percyliang — APPROVED** [review]
LGTM
Just two small comments

**Pull request merged**